pkg/nsx/services/vpc/vpc.go

```diff
@@ -473,9 +477,44 @@ func (s *VPCService) CreateorUpdateVPC(obj *v1alpha1.VPC) (*model.Vpc, *common.V
 		return nil, nil, errors.New(message)
 	}
 
+	annotations := nsObj.GetAnnotations()
+	vpcName, nameExist := annotations[common.AnnotationVPCName]
+	if nameExist {
+		log.Info("read ns annotation networkInfoName", "VPCNAME", vpcName)
+		res := strings.Split(vpcName, "/")
+		// The format should be namespace/vpc_name
+		if len(res) != 2 {
+			message := fmt.Sprintf("incorrect networkInfoName annotation %s for namespace %s", vpcName, obj.Namespace)
+			return nil, nil, errors.New(message)
+		}
+		if ns != res[0] {
+			log.Info("namespace is using shared vpc, with vpc name anno", "VPCNAME", vpcName, "Namespace", ns)
+			sharedNS := res[0]
+			sharedVPCs := s.VpcStore.GetVPCsByNamespace(sharedNS)
+			for _, vpc := range sharedVPCs {
+				if *vpc.DisplayName == res[1] {
+					return vpc, &nc, nil
+				}
+			}
+			return nil, nil, fmt.Errorf("shared vpc %s not created yet", res[1])
+		} else {
+			// temp: find vmware-system-supervisor-services-vpc has anno nsx.vmware.com/vpc_name: vmware-system-supervisor-services-vpc/vmware-system-supervisor-services
+			// this mislead shared vpc is vmware-system-supervisor-services
+			if res[1] == "infrastructure" {
+				sharedVPCName = res[1]
+			}
+		}
+	}
+
+	existingVPC := s.VpcStore.GetVPCsByNamespace(ns)
```

Comment:
you could call this method directly, it already cover the scenario if this ns is using shared vpc.

Reply:
fixed